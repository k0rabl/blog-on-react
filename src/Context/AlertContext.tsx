import React, { Component } from 'react' 

interface IState {
  isOpen: boolean
  message: string
}

const AlertContext = React.createContext({
    isOpen: false,
    message: '',
    setOpen: (c: boolean) => {},
    setType: (c: string) => {}
  })


class AlertProvider extends Component {
  // Context state
  state: IState = {
    isOpen: false,
    message: ''
  }
 

  // Method to update state
  setOpen = (isOpen: boolean) => {
    this.setState((prevState) => ({ isOpen }))
  }

  setType = (message: string) => {
    this.setState((prevState) => ({ message }))
  }

  render() {
    const { children } = this.props
    const { isOpen, message } = this.state
    const { setType, setOpen } = this

    return (
      <AlertContext.Provider
        value={{
          isOpen,
          message,
          setOpen,
          setType,
        }}
      >
        {children}
      </AlertContext.Provider>
    )
  }
}

export default AlertContext

export { AlertProvider }