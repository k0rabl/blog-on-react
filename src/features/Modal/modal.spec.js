import React from "react"
import { Modal } from  './index'
import { mount } from 'enzyme';

import { ModalProvider } from "../../Context/ModalContext";
import {
  BrowserRouter as Router,
} from "react-router-dom"


describe('<Modal />', () => {
  let component
  beforeEach(() => {
    component = mount(
      <ModalProvider value={{isOpen: true, type: 'exit'}}>
        <Router>
          <Modal />
        </Router>
      </ModalProvider>
    )
  })

  it('+++ render the component', () => {
    expect(component.find(Modal).length).toEqual(1)
    expect(component).toMatchSnapshot()
  })

  it('+++ check Button close', () => {
    component
      .find('button.btn__no')
      .simulate('click');
  })
  
  it('+++ check Button exit', () => {
    component
      .find('button.btn__exit')
      .simulate('click');
  })
  
  it('+++ check Button Delete', () => {
    component = mount(
      <ModalProvider value={{isOpen: true, type: 'delete'}}>
        <Router>
          <Modal />
        </Router>
      </ModalProvider>
    )

    component
      .find('button.btn__exit')
      .simulate('click');
  })
})