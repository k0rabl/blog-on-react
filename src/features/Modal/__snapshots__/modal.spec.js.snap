// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> +++ render the component 1`] = `
<ModalProvider
  value={
    Object {
      "isOpen": true,
      "type": "exit",
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Modal>
        <div
          className="modalC false"
        >
          <div
            className="modalC__content"
          >
            <h6
              className="modalC__title"
            >
              Exit
            </h6>
            <p
              className="modalC__text"
            >
              You realy want exit from edit-mode?
            </p>
            <div
              className="modalC__buttons buttons"
            >
              <button
                className="btn__no button"
                onClick={[Function]}
              >
                No, Thanks
              </button>
              <button
                className="btn__exit  button"
                onClick={[Function]}
              >
                Exit
              </button>
            </div>
          </div>
        </div>
      </Modal>
    </Router>
  </BrowserRouter>
</ModalProvider>
`;
